client/ladder/ladder.tsx

```diff
@@ -183,35 +210,75 @@ export function Ladder({ matchmakingType: routeType }: LadderProps) {
     [setSearchQuery],
   )
 
+  useEffect(() => {
+    dispatch(
+      getMatchmakingSeasons({
+        onSuccess: () => setLastError(undefined),
+        onError: err => setLastError(err),
+      }),
+    )
+  }, [dispatch])
+
   useEffect(() => {
     const getRankingsAbortController = new AbortController()
     const searchRankingsAbortController = new AbortController()
+    const getPastRankingsAbortController = new AbortController()
+    const searchPastRankingsAbortController = new AbortController()
     const debouncedSearch = debouncedSearchRef.current
 
-    if (searchQuery) {
-      dispatch(
-        searchRankings(matchmakingType, searchQuery, {
-          signal: searchRankingsAbortController.signal,
-          onSuccess: () => setLastError(undefined),
-          onError: err => setLastError(err),
-        }),
-      )
+    // NOTE(2Pac): Since we want to initiate the requests for retrieving the rankings and retrieving
+    // all of the seasons, including the `currentSeasonId`, at the same time (without having the
+    // waterfall requests), we're using the ref here for the `currentSeasonId` to avoid running this
+    // effect again when the `currentSeasonId` changes (e.g. from `undefined` -> actual value). It
+    // also wouldn't really make sense to run this effect again in case the `currentSeasonId` gets
+    // updated through websockets or something since that would change the rankings the user is
+    // looking at without their input.
+    if (!seasonId || !currentSeasonIdRef.current || seasonId === currentSeasonIdRef.current) {
```

Comment:
can you pull this condition out into a variable like `isForCurrentSeason` or something? Would make it easier to understand what this if statement is trying to do